src/chunkserver/scan_manager.cpp

```diff
@@ -300,14 +302,19 @@ void ScanManager::SetLocalScanMap(ScanKey key, ScanMap map) {
                      << " copysetId = " << key.second;
         return;
     }
-    if (job->currentChunkId != map.chunkid() ||
-        job->currentOffset != map.offset()) {
+    job->taskLock.RDLock();
```

Comment:
Maybe the logic of lock can be simplify: 
```
job->taskLock.RDLock();
bool matched = job->task.chunkId != map.chunkid() ||
               job->task.offset != map.offset() || 
               job->task.len != map.len();
job->taskLock.Unlock();

if (matched) {
    ...
} else {
    ...
}
```

Reply:
> Maybe the logic of lock can be simplify:
> 
> ```
> job->taskLock.RDLock();
> bool matched = job->task.chunkId != map.chunkid() ||
>                job->task.offset != map.offset() || 
>                job->task.len != map.len();
> job->taskLock.Unlock();
> 
> if (matched) {
>     ...
> } else {
>     ...
> }
> ```

done